### ISSUE TITLE
[tree view][TreeItem] Checkbox selection - Form elements do not have associated labels - Accessibility(A11y)

### ISSUE THREAD

**Issue by Matt-Butler** [mui/mui-x#15520]
### Steps to reproduce

Steps:
1. Navigate to the checkbox example for either [simple tree view](https://mui.com/x/react-tree-view/simple-tree-view/selection/#checkbox-selection) or [rich tree view](https://mui.com/x/react-tree-view/rich-tree-view/selection/#checkbox-selection).
2. Run an accessibility audit using google lighthouse.
![Screenshot 2024-11-20 at 1 15 32 PM](https://github.com/user-attachments/assets/95dc4a8c-5d28-4f90-9df9-0904a234bebd)

### Current behavior

Accessibility tests fail on checkboxes missing labels.

### Expected behavior

Accessibility tests pass

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: Tree, Accessibility

**Comment by Matt-Butler** [comment]
```
const CustomCheckbox = React.forwardRef(function CustomCheckbox({ 'aria-label': ariaLabel, checked, className, disabled, onChange, tabIndex }: CheckboxProps, ref: React.Ref<HTMLInputElement>) {
  return <Checkbox inputProps={{ 'aria-label': ariaLabel }} inputRef={ref} checked={checked} className={className} disabled={disabled} onChange={onChange} tabIndex={tabIndex} />
})

const CustomTreeItem = React.forwardRef(function CustomTreeItem(props: TreeItem2Props, ref: React.Ref<HTMLLIElement>) {
  return <TreeItem2 {...props} ref={ref} slots={{ checkbox: (props2) => <CustomCheckbox {...props2} aria-label={props.label} /> }} />
})

export function CustomCheckboxTree() {
  return (
    <Box sx={{ minHeight: 352, minWidth: 290 }}>
      <RichTreeView multiSelect checkboxSelection items={MUI_X_PRODUCTS} slots={{ item: CustomTreeItem }} />
    </Box>
  )
}
```

Here is a work around I created.

**Comment by flaviendelangle** [comment]
Thanks for the report
I can confirm that we miss an `aria-label` on the checkbox element.

The data grid sets "Select row" / "Unselect row", but we need to introduce a localization behavior to be able to do the same :cry:

**Comment by flaviendelangle** [comment]
The checkbox are not accessible through keyboard navigation, it's the item itself that gets the focus and handles the selection.
We are considering adding `aria-hidden` on the checkbox instead.

This would just require fixing the 1st point of #12654 to have a coherent experience between keyboard interactions on the item and checkbox interactions.

@LukasTy will double check if we are not missing something on a screen reader.

**Comment by abdermaiza** [comment]
@Matt-Butler 

This issue from Lighthouse is a "false positive", check this out:
https://www.codegram.com/blog/but-lighthouse-says-i-am-accessible-part-one/
https://savvasstephanides.hashnode.dev/my-lighthouse-accessibility-score-is-100-does-that-mean-my-website-is-100-accessible

:)

**Comment by LukasTy** [comment]
I confirm that adding an `aria-hidden` prop on the checkbox doesn't change anything from the screen reader's perspective.
However, I did not check if it resolves the reported lighthouse problem. 🙈

**Comment by flaviendelangle** [comment]
@LukasTy will check if `aria-hidden` solves the lighthouse issue.
If so, add `aria-hidden` with a comment saying that if we remove `tabIndex: -1` on the checkbox then we must have a proper `aria-label`.

**Comment by LukasTy** [comment]
@flaviendelangle I'm confirming that `aria-hidden="true"` on the `CheckboxItem` removes the Lighthouse error. 👍
